bundles/EcommerceFrameworkBundle/CartManager/CartItem.php

```diff
@@ -43,7 +44,8 @@ public function getSortIndex()
     public function getCart()
     {
         if (empty($this->cart)) {
-            $this->cart = Cart::getById($this->cartId);
+            $cartClass = '\\'.Factory::getInstance()->getCartManager()->getCartClassName();
+            $this->cart = $cartClass::getById($this->class);
```

Comment:
really $this->class? should that not be $this->cartId .... also phpstan tests are failing because of that. 
please double check and fix, thx.

Reply:
You are totally right, obviously a c&p failure.
I committed the change, let's see if all tests succeed.